### PR TITLE
convert EaR longevity to pipeline

### PR DESCRIPTION
**Pull request by amoskong**
We want to run EaR longevity jobs by pipeline.

http://jenkins.scylladb.com/job/enterprise-2019.1/job/EncryptionAtRest/

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: vars/EarKmipLongevityPipeline.groovy [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            string(defaultValue: '', description: '', name: 'scylla_repo')
+            string(defaultValue: "${pipelineParams.get('provision_type', 'spot_low_price')}",
+                   description: 'spot_low_price|on_demand|spot_fleet|spot_low_price|spot_duration',
+                   name: 'provision_type')
+
+            string(defaultValue: "${pipelineParams.get('post_behaviour', 'destroy')}",
+                   description: 'keep|destroy',
+                   name: 'post_behaviour')
+
+            string(defaultValue: "${pipelineParams.get('scylla_encryption_options', '\\\"{ \'cipher_algorithm\' : \'AES/ECB/PKCS5Padding\', \'secret_key_strength\' : 128, \'key_provider\': \'KmipKeyProviderFactory\', \'kmip_host\': \'kmip_test\'}\\\"')}",
```
[the remark this answers is not in the document]
Comment (amoskong) [reply vars/EarKmipLongevityPipeline.groovy]: It's flexible to make the raw parameter configurable.

Anther simple way is hardcode the encryption parameters in the code, and use a new parameter to enable encryption at rest.

```
enable_ear: kmip   / local  /  replicated
```

For current longevity test request, the second hardcode solution is enough.

File: vars/EarKmipLongevityPipeline.groovy [hunk 2]
```diff
@@ -0,0 +1,110 @@
+#!groovy
```
[the remark this answers is not in the document]
Comment (amoskong) [reply vars/EarKmipLongevityPipeline.groovy]: I don't want to add a redundant parameter to regular longevity jobs, so copied the both jenkinsfile and groovy file.

The extra yaml solution is better, currently we don't need configurable everything in jenkins job.

File: vars/EarKmipLongevityPipeline.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (amoskong) [reply vars/EarKmipLongevityPipeline.groovy]: I will save the parameter into the extra yaml.

**Comment by fruch** [comment]
@amoskong 
seems like you are missing one line ending:
https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-1311/5/execution/node/20/log/

File: configurations/kmip-ear.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
```
Comment (fruch) [inline configurations/kmip-ear.yaml]: @bentsi @roydahan 
FYI
take notice on us adding this new `configuration` folder, for the EaR cases

**Comment by amoskong** [comment]
> @amoskong
> seems like you are missing one line ending:
> https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-1311/5/execution/node/20/log/

Fixed.

**Comment by amoskong** [comment]
The PR had been tested by http://jenkins.scylladb.com/job/enterprise-2019.1/job/EncryptionAtRest/job/EaR-longevity-local-200gb-6h/15/

**Comment by amoskong** [comment]
@bentsi  ping

**Comment by bentsi** [comment]
do we have jobs in Jenkins for all the pipelines?

**Comment by amoskong** [comment]
> do we have jobs in Jenkins for all the pipelines?

Yes.

There are 6 jobs in http://jenkins.scylladb.com/job/enterprise-2019.1/job/EncryptionAtRest/
- Jobs failed for scylla issue

**Comment by amoskong** [comment]
Update:
- append a patch to rename append_conf to new name
- smoothly rebase to latest master

**Comment by amoskong** [comment]
The latest PR had been tested by http://jenkins.scylladb.com/job/enterprise-2019.1/job/EncryptionAtRest/job/EaR-longevity-kmip-200gb-6h/30/

**Comment by amoskong** [comment]
Update: rebase PR to latest master for solving the conflicts.

**Comment by fruch** [comment]
@roydahan do you approve this backport to 2019.1 ?

**Comment by roydahan** [comment]
@amoskong please solve conflicts so we can merge

**Comment by amoskong** [comment]
> @amoskong please solve conflicts so we can merge

Done.

**Pull request merged**